jupyter_notebooks/Analyzing public and hybrid cloud networks .ipynb

```diff
@@ -0,0 +1,567 @@
+    "\n",
+    "The physical network connects to the AWS network by having the exitgw establish IPSec tunnels, shown in pink, to the two transit gateways. BGP sessions running atop these tunnels to make endpoints aware of prefixes on the other side.\n",
+    "\n",
+    "You can view configuration files that we used [here](https://github.com/batfish/pybatfish/tree/master/jupyter_notebooks/networks/hybrid-cloud). The AWS portion of the configuration is in the aws_configs subfolder. It has JSON files obtained via AWS APIs. An example script that packages AWS data into a Batfish snapshot is [here](https://github.com/ratulm/bf-aws-snapshot).\n",
+    "\n",
+    "![example-bgp-network](https://raw.githubusercontent.com/batfish/pybatfish/aws-notebook/jupyter_notebooks/networks/hybrid-cloud/hybrid-cloud.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Analyzing network paths\n",
+    "\n",
+    "Batfish can help analyze cloud and hybrid networks by showing how exactly traffic flows (or not) in the network, which can help debug and fix configuration errors. Batfish can also help ensure that the network is configured exactly as desired, with respect to reachability and security policies. We illustrate these types of analysis below.\n",
+    "\n",
+    "First, lets define a couple of maps to help with the analysis."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Instances in AWS in each region and VPC type (public, private)\n",
+    "hosts = {}\n",
+    "hosts[\"east2_private\"] = \"i-04cd3db5124a05ee6\"\n",
+    "hosts[\"east2_public\"] = \"i-01602d9efaed4409a\"\n",
+    "hosts[\"west2_private\"] = \"i-0a5d64b8b58c6dd09\"\n",
+    "hosts[\"west2_public\"] = \"i-02cae6eaa9edeed70\"\n",
+    "\n",
+    "#Public IPs of instances in AWS\n",
+    "public_ips = {}\n",
+    "public_ips[\"east2_public\"] = \"13.59.144.125\" # of i-01602d9efaed4409a\n",
+    "public_ips[\"west2_public\"] = \"54.191.42.182\" # of i-02cae6eaa9edeed70"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Paths across VPCs within an AWS region\n",
+    "\n",
+    "To see how traffic flows between two instances in the same region but across different VPCs, say, from `hosts[\"east2_private\"]` to `hosts[\"east2_public\"]`, we can run a traceroute query across them as follows. \n",
+    "\n",
+    "In the query below, we use the name of the instance as the destination for the traceroute. This makes Batfish pick the instance's private (i.e., non-Elastic) IP (`10.20.1.207`). It does not pick the public IP because that those IPs do not reside on instances but are used by the Internet gateway to NAT instance's traffic in and out of AWS. If an instance has multiple private IPs, Batfish will pick one at random. To make Batfish use a specific IP, supply that IP as the argument to the `dstIps` parameter."
```

Comment:
Is there an AWS doc that explains that public IP are not used by default when referring to an instance by name?